tinygrad/codegen/uopgraph.py

```diff
@@ -553,13 +557,16 @@ def get_recursive_children(x:UOp, end:UOps, include_self=False) -> Set[UOp]:
 
     # scope children impact the toposort and END* insertion
     scope_children = {p:get_recursive_children(p, END_FOR_UOP[p.op][0]) for p in reversed(in_degree) if p.op in END_FOR_UOP}
+    range_phi = {r:[p for p in scope_children[r] if p.op is UOps.PHI] for r in scope_children if r.op is UOps.RANGE}
+    range_srcs = {p:range_srcs[p] for p in range_srcs if p.op is UOps.PHI}
 
     queue:List[Tuple[int, UOp]] = []
     def push(u:UOp):
       priority = 0
       # prefer uops that are loop children
-      for l, ss in scope_children.items():
-        if l.op is UOps.RANGE and u in ss: priority -= l.arg[0]*1000 + l.arg[1]
+      if u.op is UOps.RANGE and u.arg[1]: priority += u.arg[0] + 10000 * \
```

Comment:
Feel free to use more lines for this and add comments. I don't understand what this is doing

Reply:
I changed it to look like this:
```
if u.op is UOps.RANGE and u.arg[1]: 
        priority += u.arg[0]
        for p in range_phi[u]: # for every phi in this ranges scope
          # add up all the ranges that precede it
          # except those that are also part of this reduction
          priority += 10000*len([r for r in range_srcs[p] if not set(range_phi[u]).intersection(set(range_phi[r]))])
```

is that more clear?

Reply:
there are other ways to do the intersection if we don't want to use sets, I just used it here because it more intuitive to what's actually being calculated